core/nightshade/src/nightshade.rs

```diff
@@ -0,0 +1,781 @@
+/// Nightshade v2
+use std::cmp::{max, min, Ordering};
+use std::collections::HashSet;
+
+use serde::Serialize;
+
+use primitives::aggregate_signature::{AggregatePublicKey, BlsAggregateSignature, BlsPublicKey, BlsSecretKey, BlsSignature};
+use primitives::hash::{CryptoHash, hash_struct};
+use primitives::serialize::Encode;
+use primitives::signature::bs58_serializer;
+
+pub type AuthorityId = usize;
+
+const COMMIT_THRESHOLD: i64 = 3;
+
+pub enum NSResult {
```

Comment:
Should we just use Result here? Given it's only 2 states

Reply:
I planned to extend it, but after all, I keep only these two fields. I'll use Result instead

Reply:
Can just make it `pub type NSResult = Result<Option<State>, String>;`

Reply:
done